docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,505 @@
+This is where |sr-long| helps: it provides centralized schema management and compatibility checks as schemas evolve.
+
+For a more in-depth understanding of the benefits of Avro and |sr|, please refer to the following resources:
+
+* `Why Avro For Kafka Data <https://www.confluent.io/blog/avro-kafka-data/>`_
+* `Yes, Virginia, You Really Do Need a Schema Registry <https://www.confluent.io/blog/schema-registry-kafka-stream-processing-yes-virginia-you-really-need-one/>`_
+
+Target Audience
+^^^^^^^^^^^^^^^
+
+The target audience is a developer writing Kafka streaming applications, who wants to build a robust application leveraging Avro data and |sr-long|.
+
+This tutorial is not meant to cover the operational aspects of running the |sr| service. For production deployments of |sr-long|, please refer to :ref:`Schema Registry Operations<schemaregistry_operations>`.
+
+Before You Begin
+~~~~~~~~~~~~~~~~
+
+Prerequisites
+^^^^^^^^^^^^^
+
+Before proceeding with this tutorial
+
+1. Verify have installed on your local machine:
+
+* Java 1.8 to run |cp|
+* Maven to compile the client Java code
+* `jq` tool to make output from SR REST endpoint pretty
+
+2. Use the :ref:`quickstart` to bring up |cp|. With a single-line command, you can have running on your local machine a basic Kafka cluster with |sr-long| and other services.
+
+.. sourcecode:: bash
+
+   $ confluent start
+
+   This CLI is intended for development only, not for production
+   https://docs.confluent.io/current/cli/index.html
+
+   Starting zookeeper
+   zookeeper is [UP]
+   Starting kafka
+   kafka is [UP]
+   Starting schema-registry
+   schema-registry is [UP]
+   Starting kafka-rest
+   kafka-rest is [UP]
+   Starting connect
+   connect is [UP]
+   Starting ksql-server
+   ksql-server is [UP]
+   Starting control-center
+   control-center is [UP]
+
+
+3. Clone the |cp| `examples` repo in GitHub and work in the `clients/avro/` subdirectory.
+
+.. sourcecode:: bash
+
+   $ git clone https://github.com/confluentinc/examples.git
+   $ cd examples/clients/avro
+
+
+.. _schema_registry_tutorial_definition:
+
+Terminology Levelset
+^^^^^^^^^^^^^^^^^^^^
+
+First let us levelset on terminology: what is a `schema` versus a `topic` versus a `subject`.
+
+A Kafka topic contains messages, and each message is a key-value pair.
+Either the message key or the message value, or both, can be independently serialized as Avro.
+The Kafka topic name is independent of the schema name.
+When a producer writes a message to a Kafka topic, it can serialize the message key or message value as Avro (or both).
+By default, the subject that is registered in |sr| is derived from the Kafka topic name.
+
+As a practical example, let's say a retail business is streaming transactions in a Kafka topic called `transactions`.
+A producer is writing data with a schema `Payment` to that Kafka topic.
+If the producer is serializing the message value as Avro, |sr| has a subject called `transactions-value`.
```

Comment:
Need double-ticks for literals: ``transactions-value``

Reply:
Same for \`\`transactions-key\`\` below